app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -38,287 +35,134 @@ async function addEthereumChainHandler(
-        }),
-      );
-    }
-
-    if (!nativeCurrency.symbol || typeof nativeCurrency.symbol !== 'string') {
-      return end(
-        ethErrors.rpc.invalidParams({
-          message: `Expected a string 'nativeCurrency.symbol'. Received: ${nativeCurrency.symbol}`,
-        }),
+      networkConfigurationId = await upsertNetworkConfiguration(
+        {
+          chainId,
+          rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
+          nickname: chainName,
+          rpcUrl: firstValidRPCUrl,
+          ticker,
+        },
+        { source: 'dapp', referrer: origin },
       );
+    } catch (error) {
+      endApprovalFlow({ id: approvalFlowId });
+      return end(error);
     }
-  }
 
-  const ticker = nativeCurrency?.symbol || UNKNOWN_TICKER_SYMBOL;
-
-  if (
-    ticker !== UNKNOWN_TICKER_SYMBOL &&
-    (typeof ticker !== 'string' || ticker.length < 2 || ticker.length > 6)
-  ) {
-    return end(
-      ethErrors.rpc.invalidParams({
-        message: `Expected 2-6 character string 'nativeCurrency.symbol'. Received:\n${ticker}`,
-      }),
-    );
-  }
-  // if the chainId is the same as an existing network but the ticker is different we want to block this action
-  // as it is potentially malicious and confusing
-  if (
-    existingNetwork &&
-    existingNetwork.chainId === _chainId &&
-    existingNetwork.ticker !== ticker
-  ) {
-    return end(
-      ethErrors.rpc.invalidParams({
-        message: `nativeCurrency.symbol does not match currency symbol for a network the user already has added with the same chainId. Received:\n${ticker}`,
-      }),
-    );
-  }
-  let networkConfigurationId;
-
-  const { id: approvalFlowId } = await startApprovalFlow();
-
-  try {
-    await requestUserApproval({
-      origin,
-      type: ApprovalType.AddEthereumChain,
-      requestData: {
-        chainId: _chainId,
-        rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
-        chainName: _chainName,
-        rpcUrl: firstValidRPCUrl,
-        ticker,
-      },
-    });
-
-    networkConfigurationId = await upsertNetworkConfiguration(
-      {
-        chainId: _chainId,
-        rpcPrefs: { blockExplorerUrl: firstValidBlockExplorerUrl },
-        nickname: _chainName,
+    requestData = {
+      toNetworkConfiguration: {
         rpcUrl: firstValidRPCUrl,
+        chainId,
+        nickname: chainName,
         ticker,
+        networkConfigurationId,
       },
-      { source: MetaMetricsNetworkEventSource.Dapp, referrer: origin },
-    );
-
-    // Once the network has been added, the requested is considered successful
-    res.result = null;
-  } catch (error) {
-    endApprovalFlow({ id: approvalFlowId });
-    return end(error);
-  }
+      fromNetworkConfiguration: currentNetworkConfiguration,
+    };
+  } else {
+    networkConfigurationId = existingNetwork.id ?? existingNetwork.type;
```

Comment:
Also following https://github.com/MetaMask/core/pull/4286, the network configuration won't have an `id` property; rather, RPC endpoints will have an associated `networkClientId`.